dev-desktop: Clang can't link C++ code

Tested on dev-desktop-us-1. @jackh726 was trying to use the wild linker on a dev desktop and hit an error.

After some investigation, the problem appears to be due to clang trying to use gcc-14 libraries, but the c++ library is only installed for gcc-13.

The problem can be reproduced as follows:

```sh
echo 'int main() {}' | clang++ -x c++ - -o /dev/null
/usr/bin/ld: cannot find -lstdc++: No such file or directory
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

I tried setting up an ubuntu 24.04 arm64 docker image and the problem didn't reproduce. However, after installing `libgcc-14-dev` the problem did reproduce. Further installing `libstdc++-14-dev` then fixed it. So I guess the fix is either to uninstall packages that put stuff in `/usr/lib/gcc/aarch64-linux-gnu/14` so that clang doesn't try to use it or install `libstdc++-14-dev` or something that depends on it, perhaps `g++-14-aarch64-linux-gnu`.

I was curious as to why libgcc-14-dev was installed. Seems that it's because of emacs.

```
aptitude why libgcc-14-dev
i A emacs-gtk  Provides emacs             
i A emacs-gtk  Depends  libgccjit0 (>= 12)
i A libgccjit0 Depends  libgcc-14-dev     
```

So I think installing `g++-14-aarch64-linux-gnu` is the best option. I'll send a PR.

--- a/ansible/roles/dev-desktop/tasks/dependencies.yml
+++ b/ansible/roles/dev-desktop/tasks/dependencies.yml
@@ -17,6 +17,7 @@
       - cmake
       - direnv
       - fzf
+      - g++-14-aarch64-linux-gnu # Required for functional clang++ when libgcc-14-dev is installed
       - gcc-mingw-w64-x86-64 # Allows running `x check --target x86_64-pc-windows-gnu`
       - jq
       - tidy # Format and diff html for rustdoc

